### ISSUE TITLE
[PHP 8.1] Add Dummy class to ENUM migration

### ISSUE THREAD

**Issue by TomasVotruba** [rectorphp/rector#5395]
- RFC: https://wiki.php.net/rfc/enumerations

## Todo

- [x] add to https://github.com/nikic/PHP-Parser
- [x] identify possible input pattern

## Most used packages in PHP userland

Asked: https://twitter.com/rectorphp/status/1388876287477944325

- https://github.com/myclabs/php-enum
- https://github.com/spatie/enum
- https://github.com/marc-mabe/php-enum
- https://github.com/Elao/PhpEnums

**Comment by TomasVotruba** [comment]
Ref https://github.com/nikic/PHP-Parser/pull/758

**Comment by TomasVotruba** [comment]
Not priority ATM, this might take a while to merge in php-parser

**Comment by TomasVotruba** [comment]
The got merged  nikic/PHP-Parser#758 :heavy_check_mark:

**Comment by aLmonndo** [comment]
Quote from "Tomasz Kowalczyk" on LinkedIn:

_"Platenum: https://github.com/thunderer/Platenum. I analyzed many popular enum libraries on Packagist with a detailed benchmark here: https://github.com/thunderer/PhpEnumerations. There was no single library that provided desired guarantees (truly single instance, strict equality, etc.) and complete feature set, that's why I wrote one which does."_

Quote from "Petru Szemereczki" on FB:

_"this one got mentioned in symfony's issue tracking: https://github.com/Elao/PhpEnums/issues/124
in package's readme they mention other 3 libs
but mostly, my "enum" are only plain php class / interface that has public const with either UPPERCASE_SNAKE or (since 8.1 enum rfc) UpperCamelCase. and in code i reference the actual data type behind an enum (mostly string) and use either JetBrains' ExpectedValues attribute or psalm's @psalm-xxx MyEnum::* (or both).
another personal take would be to have class constants + an constant array of valid values and some static methods that ensures the validity during runtime: https://github.com/hktr92/oopize/blob/master/src/Util/EnumUtil.php?fbclid=IwAR24p3v3HasVOzVN24l2_ucfAbdsCLSBjSK02RWY1qbtcnZxch-UUXwVGd4
it's used mostly by very few projects I've made. the idea was fine back in php 7, but now it doesn't really make any sense to have / use such a library.
i personally won't make such a feature in rector because it's such a wide possibility to exist."_

"Quote from "Tobias Pogadl" on FB:

_"if you don’t have any yet, think about to skip this for a few months or only choosing one that is compatible with enums in php 8.1"_
